conan/cli/commands/install.py

```diff
@@ -109,21 +42,40 @@ def install(conan_api, parser, *args):
     if args.requires and (args.name or args.version or args.user or args.channel):
         raise ConanException("Can't use --name, --version, --user or --channel arguments with "
                              "--requires")
-
+    if not args.path and not args.requires and not args.tool_requires:
+        raise ConanException("Please specify at least a path to a conanfile or a valid reference.")
     cwd = os.getcwd()
+
     if args.path:
         path = conan_api.local.get_conanfile_path(args.path, cwd, py=None)
         source_folder = os.path.dirname(path)
     else:
         source_folder = cwd
+        path = None
     if args.output_folder:
         output_folder = make_abs_path(args.output_folder, cwd)
     else:
         output_folder = None
 
+    # Basic collaborators, remotes, lockfile, profiles
     remotes = conan_api.remotes.list(args.remote)
-
-    deps_graph, lockfile = graph_compute(args, conan_api, partial=args.lockfile_partial)
+    lockfile = conan_api.lockfile.get_lockfile(lockfile=args.lockfile,
+                                               conanfile_path=path,
+                                               cwd=cwd,
+                                               partial=args.lockfile_partial)
+    profile_host, profile_build = conan_api.profiles.get_profiles_from_args(args)
+    if path:
+        deps_graph = conan_api.graph.load_graph_consumer(path, args.name, args.version,
+                                                         args.user, args.channel,
+                                                         profile_host, profile_build, lockfile,
+                                                         remotes, args.update)
+    else:
+        deps_graph = conan_api.graph.load_graph_requires(args.requires, args.tool_requires,
```

Comment:
👍  This is much better then and giant function above

Reply:
https://github.com/prince-chrismc/conan-center-index/commit/1909a3e875fc9b35ffb5c5b00080ce9de087f7be

This is much better for UX 
![image](https://user-images.githubusercontent.com/16867443/207449353-3a7d21c4-dff2-4132-aa09-df92641c3410.png)